### PR TITLE
Initial support for creating dashboards from metrics/alarms.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Currently untested.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
This is ready for review.

File: sdk/nodejs/cloudwatch/widgets_json.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+import * as pulumi from "@pulumi/pulumi";
+
+import { YAxis } from "./widgets_graph";
+
+// All the typings for the json blob we need to create for Dashboard.dashboardBody. We'll convert
+// the right in-memory auto-laid-out system we have into these.  These typings are purely so we do
+// the json conversion properly as it's very easy to screw up if using things like the 'any' type.
+
+export interface WidgetJson {
```
Comment (ekrengel) [inline sdk/nodejs/cloudwatch/widgets_json.ts]: I like that you moved all of this into its own file. May do something similar w the Swagger JSON typings.

File: sdk/nodejs/cloudwatch/widgets_simple.ts [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+     */
+    height?: number;
+}
+
+/**
+ * Base type of all non-flow Widgets to place in a DashboardGrid.
+ */
+export abstract class SimpleWidget implements Widget {
+    constructor(private readonly args: SimpleWidgetArgs) {
+        if (args.width > 24) {
```
Comment (ekrengel) [inline sdk/nodejs/cloudwatch/widgets_simple.ts]: Should you also check the min value here?

**Comment by CyrusNajmabadi** [comment]
Closing out.  This came through https://github.com/pulumi/pulumi-awsx/pull/217

**Pull request closed without merging**